src/astro/utils/table_handler.py

```diff
@@ -16,28 +16,46 @@ def _set_variables_from_first_table(self):
         """
         first_table: Optional[Table] = None
         if self.op_args:
-            table_index = [x for x, t in enumerate(self.op_args) if type(t) == Table]
-            if table_index:
+            table_index = [x for x, t in enumerate(self.op_args) if type(t) is Table]
+            conn_id_set = {x.conn_id for x in self.op_args if type(x) is Table}
+            # Check to see if all tables belong to same conn_id. Otherwise, we this can go wrong for cases
+            # 1. When we have tables from different DBs.
+            # 2. When we have tables from different conn_id, since they can be configured with different
+            # database/schema etc.
+            if table_index and len(conn_id_set) == 1:
```

Comment:
If this condition is not met, should we raise an exception? Should we have a test to cover if the right exception is raised?

Reply:
I think it's a valid case if you have tables from two different connections. We just can't select DB-related params from the first table in that case.